core/src/main/java/com/yahoo/ycsb/workloads/CoreWorkload.java

```diff
@@ -498,28 +500,35 @@ else if (op.compareTo("SCAN")==0)
 		return true;
 	}
 
-    int nextKeynum() {
-        int keynum;
-        if(keychooser instanceof ExponentialGenerator) {
+    long nextKeynum() {
+        long keynum=-1;
+
+        if(keychooser instanceof ExponentialLongGenerator) {
             do
                 {
-                    keynum=transactioninsertkeysequence.lastInt() - keychooser.nextInt();
+                    //keynum=transactioninsertkeysequence.lastLong() - keychooser.nextInt();
```

Comment:
I started getting this merged in, but testing failed. looks like having this commented out results in an infinite loop.